ci/kokoro/install/build.sh

```diff
@@ -49,46 +49,59 @@ else
   exit 1
 fi
 
-echo "================================================================"
-echo "Load Google Container Registry configuration parameters $(date)."
-
-if [[ -f "${KOKORO_GFILE_DIR:-}/gcr-configuration.sh" ]]; then
-  source "${KOKORO_GFILE_DIR:-}/gcr-configuration.sh"
-fi
-
 if [[ -z "${PROJECT_ROOT+x}" ]]; then
   readonly PROJECT_ROOT="$(
     cd "$(dirname "$0")/../../.."
     pwd
   )"
 fi
+GCLOUD=gcloud
 source "${PROJECT_ROOT}/ci/kokoro/define-docker-variables.sh"
+source "${PROJECT_ROOT}/ci/colors.sh"
+source "${PROJECT_ROOT}/ci/kokoro/gcloud-functions.sh"
+source "${PROJECT_ROOT}/ci/kokoro/cache-functions.sh"
 
 echo "================================================================"
-echo "Change working directory to project root $(date)."
+log_yellow "Change working directory to project root."
 cd "${PROJECT_ROOT}"
 
 echo "================================================================"
-echo "Building with ${NCPU} cores $(date) on ${PWD}."
+log_normal "Load Google Container Registry configuration parameters."
+
+if [[ -f "${KOKORO_GFILE_DIR:-}/gcr-configuration.sh" ]]; then
+  source "${KOKORO_GFILE_DIR:-}/gcr-configuration.sh"
+fi
+
+echo "================================================================"
+log_normal "Building with ${NCPU} cores on ${PWD}."
 
 echo "================================================================"
-echo "Setup Google Container Registry access $(date)."
+log_normal "Setup Google Container Registry access."
 if [[ -f "${KOKORO_GFILE_DIR:-}/gcr-service-account.json" ]]; then
   gcloud auth activate-service-account --key-file \
     "${KOKORO_GFILE_DIR}/gcr-service-account.json"
 fi
 gcloud auth configure-docker
 
 echo "================================================================"
-echo "Download existing image (if available) for ${DISTRO} $(date)."
+log_normal "Download existing image (if available) for ${DISTRO}."
 has_cache="false"
 if docker pull "${INSTALL_IMAGE}:latest"; then
   echo "Existing image successfully downloaded."
   has_cache="true"
 fi
 
+readonly CACHE_BUCKET="${GOOGLE_CLOUD_CPP_KOKORO_RESULTS:-cloud-cpp-kokoro-results}"
+readonly CACHE_FOLDER="${CACHE_BUCKET}/build-cache/google-cloud-cpp/master/install"
+readonly CACHE_NAME="${DISTRO}.tar.gz"
+
+if cache_download_enabled; then
+  cache_download_tarball \
+    "${CACHE_FOLDER}" "ci/kokoro/install/ccache-contents" "${CACHE_NAME}" || true
```

Comment:
It seems a little strange that we're downloading ephemeral, working files and stuff to a directory inside our repo. Is there a reason why this needs to be the case? Can this not be somewhere like /tmp or something? It's fine to keep, but I'm just curious the reason.

Reply:
FTR, @coryan and I discussed this offline. It would be nice to avoid having a cache directory in our repo, but neither of us can think of a reasonable way to avoid it at this point.

Reply:
Created #3915